frontend/pages/workspaces/[workspaceId]/folders/[folderId]/index.tsx

```
@@ -85,6 +88,23 @@ const FolderHomepage: NextPage = () => {
     variables: { folder: folderId },
   });
 
+  const [returnedUserPermissons] = useGetWorkspaceMembershipQuery({
```

I think this way of checking permissions is not enough. The query `useGetFolderByIdQuery` must not return any folder information if the user doesn't have the permissions to see it. Otherwise we leak the folder data over the API.

---

Sorry, that I didn't see this earlier.

I'm not sure if this query makes sense then. At least for this usecase, we could possibly look at the returned type/error of the `GetFolderById` query and use that to determine if we have access.

I'm not sure if `GetFolderById` should return a big enum with 2 cases:
- You have access (here is the folder data)
- You don't have access

Or if `﻿GetFolderById` should return an error. Apparently errors can be annotated with additional properties. So we don't have to make the decision based on the error message alone:
- https://async-graphql.github.io/async-graphql/en/error_handling.html
- https://async-graphql.github.io/async-graphql/en/error_extensions.html